ui/apps/akasha/src/components/pages/entry-page/beam-page.tsx

```diff
@@ -106,48 +106,41 @@ const BeamPage: React.FC<BeamPageProps> = props => {
             </>
           }
           queryKey={`reflect-feed-${beamId}`}
-          filters={{ where: { reflection: { isNull: true } } }}
+          filters={{ where: { beamID: { equalTo: beamId } } }}
           estimatedHeight={120}
-          renderItem={itemData => (
-            <>
-              <Divider />
-              <EditableReflection
-                entryData={mapReflectEntryData(itemData.node)}
-                reflectToId={beamId}
-                contentClickable={true}
-                onContentClick={() =>
-                  navigate({
-                    to: '/reflection/$reflectionId',
-                    params: {
-                      reflectionId: itemData.node.id,
-                    },
-                  })
-                }
-                onReflect={() =>
-                  navigate({
-                    to: '/reflection/$reflectionId/reflect',
-                    params: {
-                      reflectionId: itemData.node.id,
-                    },
-                  })
-                }
-              />
-              <ReflectionPreview
-                reflectionId={itemData.node.id}
-                onNavigate={(options: { id: string; reflect?: boolean }) => {
-                  navigate({
-                    to: options.reflect
-                      ? '/reflection/$reflectionId/reflect'
-                      : '/reflection/$reflectionId',
-                    params: {
-                      reflectionId: options.id,
-                    },
-                  });
-                  return;
+          renderItem={itemData => {
+            return (
+              <ErrorBoundary
+                errorObj={{
+                  type: 'script-error',
+                  title: t('Error in loading reflection.'),
                 }}
-              />
-            </>
-          )}
+                logger={logger}
+              >
+                <>
+                  <Divider />
+                  <EditableReflectionResolver
+                    reflectID={itemData.node.reflectionID}
+                    beamID={beamId}
+                  />
```

Comment:
Need to expose the `reflectToId` for this component and set it as `beamId` otherwise it will not be correct. Can you check this closely ?